core/variant_op.cpp

```diff
@@ -161,35 +162,78 @@ bool Variant::booleanize(bool &r_valid) const {
 			return _data._real;
 		case STRING:
 			return (*reinterpret_cast<const String *>(_data._mem)) != "";
-		case VECTOR2:
-		case RECT2:
-		case TRANSFORM2D:
-		case VECTOR3:
-		case PLANE:
-		case RECT3:
-		case QUAT:
-		case BASIS:
-		case TRANSFORM:
-		case COLOR:
+		case VECTOR2: {
+			const Vector2 *v = reinterpret_cast<const Vector2 *>(_data._mem);
+			return v->x + v->y;
```

Comment:
As discussed on IRC, this would make `Vector2(1, -1)` or `Vector2(-42, 42)` evaluate to `true`. The logic should be: false if all components are 0, else true, so `v->x || v->y`.